### ISSUE TITLE
ERR_INVALID_ARG_TYPE after android@7.1.2 update

### ISSUE THREAD

**Issue by uerceg** [apache/cordova-android#554]
Hey guys.

We have our plugin built for Cordova and the version on Github `master` branch is compatible with pre `android@7.0.0` platform version. For `android@7.0.0` and above we made separate branch called `cordova-8` from which people can add our plugin in case they are using those platform version(s).

Together with `android@7.1.1` platform version, everything worked fine, but as of `android@7.1.2`, plugin can't be added on Android platform. I was trying to go through all the PRs you merged as part of `android@7.1.2` update, but no luck in finding what is actually causing the error.

Error which happens once plugin is being added:

```
No scripts found for hook "before_plugin_add".
Calling plugman.fetch on plugin "https://github.com/adjust/cordova_sdk.git#cordova-8"
saving
Running command: npm install https://github.com/adjust/cordova_sdk.git#cordova-8 --production --save
Command finished with error code 0: npm install,https://github.com/adjust/cordova_sdk.git#cordova-8,--production,--save
Copying plugin "/Users/uerceg/Projects/Cordova/AdjustCordovaEight/node_modules/com.adjust.sdk" => "/Users/uerceg/Projects/Cordova/AdjustCordovaEight/plugins/com.adjust.sdk"
Calling plugman.install on plugin "/Users/uerceg/Projects/Cordova/AdjustCordovaEight/plugins/com.adjust.sdk" for platform "android
Installing "com.adjust.sdk" for android
Running command: /Users/uerceg/Projects/Cordova/AdjustCordovaEight/platforms/android/cordova/version
Command finished with error code 0: /Users/uerceg/Projects/Cordova/AdjustCordovaEight/platforms/android/cordova/version
Finding scripts for "before_plugin_install" hook from plugin com.adjust.sdk on android platform only.
No scripts found for hook "before_plugin_install".
Install start for "com.adjust.sdk" on android.
PlatformApi successfully found for platform android
Android Studio project detected
Beginning processing of action stack for android project...
Error during processing of action! Attempting to revert...
Failed to install 'com.adjust.sdk': TypeError [ERR_INVALID_ARG_TYPE]: Uh oh!
The "path" argument must be of type string
    at assertPath (path.js:28:11)
    at Object.resolve (path.js:1179:7)
    at copyNewFile (/Users/uerceg/Projects/Cordova/AdjustCordovaEight/platforms/android/cordova/lib/pluginHandlers.js:258:28)
    at install (/Users/uerceg/Projects/Cordova/AdjustCordovaEight/platforms/android/cordova/lib/pluginHandlers.js:43:17)
    at ActionStack.process (/Users/uerceg/Projects/Cordova/AdjustCordovaEight/platforms/android/cordova/node_modules/cordova-common/src/ActionStack.js:56:25)
    at PluginManager.doOperation (/Users/uerceg/Projects/Cordova/AdjustCordovaEight/platforms/android/cordova/node_modules/cordova-common/src/PluginManager.js:114:20)
    at PluginManager.addPlugin (/Users/uerceg/Projects/Cordova/AdjustCordovaEight/platforms/android/cordova/node_modules/cordova-common/src/PluginManager.js:144:17)
    at /Users/uerceg/Projects/Cordova/AdjustCordovaEight/platforms/android/cordova/Api.js:247:74
    at _fulfilled (/Users/uerceg/Projects/Cordova/AdjustCordovaEight/platforms/android/cordova/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (/Users/uerceg/Projects/Cordova/AdjustCordovaEight/platforms/android/cordova/node_modules/q/q.js:863:30)
Uh oh!
The "path" argument must be of type string
TypeError [ERR_INVALID_ARG_TYPE]: Uh oh!
The "path" argument must be of type string
    at assertPath (path.js:28:11)
    at Object.resolve (path.js:1179:7)
    at copyNewFile (/Users/uerceg/Projects/Cordova/AdjustCordovaEight/platforms/android/cordova/lib/pluginHandlers.js:258:28)
    at install (/Users/uerceg/Projects/Cordova/AdjustCordovaEight/platforms/android/cordova/lib/pluginHandlers.js:43:17)
    at ActionStack.process (/Users/uerceg/Projects/Cordova/AdjustCordovaEight/platforms/android/cordova/node_modules/cordova-common/src/ActionStack.js:56:25)
    at PluginManager.doOperation (/Users/uerceg/Projects/Cordova/AdjustCordovaEight/platforms/android/cordova/node_modules/cordova-common/src/PluginManager.js:114:20)
    at PluginManager.addPlugin (/Users/uerceg/Projects/Cordova/AdjustCordovaEight/platforms/android/cordova/node_modules/cordova-common/src/PluginManager.js:144:17)
    at /Users/uerceg/Projects/Cordova/AdjustCordovaEight/platforms/android/cordova/Api.js:247:74
    at _fulfilled (/Users/uerceg/Projects/Cordova/AdjustCordovaEight/platforms/android/cordova/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (/Users/uerceg/Projects/Cordova/AdjustCordovaEight/platforms/android/cordova/node_modules/q/q.js:863:30)
```

Steps to reproduce:

1. Create empty new Cordova app.
2. `cordova platform add android@7.1.2`
3. `cordova plugin add https://github.com/adjust/cordova_sdk.git\#cordova-8`

Testing environment:

- Mac OS X Mojave 10.14
- Cordova version: `8.1.2 (cordova-lib@8.1.1)`

Plugin code is publicly available on Github.

In case you need any additional info, please feel free to ping.

Looking forward to hear from you soon.

Cheers

**Comment by brodybits** [comment]
Already solved in `master`, tested as follows:

* `cordova platform add cordova-android@nightly`
* `cordova plugin add https://github.com/adjust/cordova_sdk.git\#cordova-8`
* `cordova build android`

Will be fixed in 7.1.3. Closing as a duplicate of GH-540.

**Comment by uerceg** [comment]
Awesome guys, thanks for super quick reply.

I tried @brodybits's flow with nightly build and it works like a charm. Nice. 👍 

I am just not sure I understood P.S. comment. Is there a way to declare paths in `plugin.xml` in a way that it supports both, `cordova-android@6` and `cordova-android@7`?

**Comment by brodybits** [comment]
> Is there a way to declare paths in `plugin.xml` in a way that it supports both, `cordova-android@6` and `cordova-android@7`?

Once PR #550 is merged, it should just work if you revert <https://github.com/adjust/cordova_sdk/commit/114fb7671b405b0b19c251ca81c10c7a0c20ea0a>.

PR #550 was raised due to the large number of plugins that were broken by the 7.0.0 release.

**Comment by uerceg** [comment]
Awesome. Thank you very much, will keep an eye on 7.1.3 official release and update plugin and docs accordingly.

Cheers! 🍺
